[BUG] Enums are not being deserialized correctly for all plugins

**Describe the bug**
Due to some jackson dependabot update, but deserializing Enums does not work in `data-prepper-core` for all plugins now

I noticed this when Enum deserialization for the opensearch source mysteriously started failing with 

```
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Cannot construct instance of `java.lang.Enum` (no Creators, like default constructor, exist): abstract types either need to be mapped to concrete types, have custom deserializer, or contain additional type information
``` 

for the `search_context_type`. I manually fixed that but the http sink `http_method` also does not work (not sure if it worked before). There may be other places where enums would fail to deserialize too now

**To Reproduce**
Steps to reproduce the behavior:
Create a pipelines with http sink and `http_method` will not deserialize

```
sink:
  - http:
      http_method: "POST"
      url: "test-url"
```

Same fails for sqs `notification_type`


**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Environment (please complete the following information):**
 - OS: [e.g. Ubuntu 20.04 LTS]
 - Version [e.g. 22]

**Additional context**
Add any other context about the problem here.

--- a/data-prepper-core/src/main/java/org/opensearch/dataprepper/plugin/ObjectMapperConfiguration.java
+++ b/data-prepper-core/src/main/java/org/opensearch/dataprepper/plugin/ObjectMapperConfiguration.java
@@ -17,7 +17,7 @@ import java.util.Set;
 public class ObjectMapperConfiguration {
     static final Set<Class> TRANSLATE_VALUE_SUPPORTED_JAVA_TYPES = Set.of(
             String.class, Number.class, Long.class, Short.class, Integer.class, Double.class, Float.class,
-            Boolean.class, Duration.class, Enum.class, Character.class);
+            Boolean.class, Character.class);
 
     @Bean(name = "extensionPluginConfigObjectMapper")
     ObjectMapper extensionPluginConfigObjectMapper() {
@@ -32,6 +32,7 @@ public class ObjectMapperConfiguration {
     @Bean(name = "pluginConfigObjectMapper")
     ObjectMapper pluginConfigObjectMapper(final VariableExpander variableExpander) {
         final SimpleModule simpleModule = new SimpleModule();
+        simpleModule.addDeserializer(Duration.class, new DataPrepperDurationDeserializer());
         TRANSLATE_VALUE_SUPPORTED_JAVA_TYPES.stream().forEach(clazz -> simpleModule.addDeserializer(
                 clazz, new DataPrepperScalarTypeDeserializer<>(variableExpander, clazz)));
 

--- a/data-prepper-core/src/test/java/org/opensearch/dataprepper/plugin/ObjectMapperConfigurationTest.java
+++ b/data-prepper-core/src/test/java/org/opensearch/dataprepper/plugin/ObjectMapperConfigurationTest.java
@@ -1,0 +1,81 @@
+package org.opensearch.dataprepper.plugin;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+@ExtendWith(MockitoExtension.class)
+class ObjectMapperConfigurationTest {
+    private final ObjectMapperConfiguration objectMapperConfiguration = new ObjectMapperConfiguration();
+
+    @Mock
+    private VariableExpander variableExpander;
+
+    @Test
+    void test_duration_with_pluginConfigObjectMapper() {
+        final String durationTestString = "10s";
+        final ObjectMapper objectMapper = objectMapperConfiguration.pluginConfigObjectMapper(variableExpander);
+        final Duration duration = objectMapper.convertValue(durationTestString, Duration.class);
+        assertThat(duration, equalTo(Duration.ofSeconds(10)));
+    }
+
+    @Test
+    void test_enum_with_pluginConfigObjectMapper() {
+        final String testString = "test";
+        final ObjectMapper objectMapper = objectMapperConfiguration.pluginConfigObjectMapper(variableExpander);
+        final TestType duration = objectMapper.convertValue(testString, TestType.class);
+        assertThat(duration, equalTo(TestType.fromOptionValue(testString)));
+    }
+
+    @Test
+    void test_duration_with_extensionPluginConfigObjectMapper() {
+        final String durationTestString = "10s";
+        final ObjectMapper objectMapper = objectMapperConfiguration.extensionPluginConfigObjectMapper();
+        final Duration duration = objectMapper.convertValue(durationTestString, Duration.class);
+        assertThat(duration, equalTo(Duration.ofSeconds(10)));
+    }
+
+    @Test
+    void test_enum_with_extensionPluginConfigObjectMapper() {
+        final String testString = "test";
+        final ObjectMapper objectMapper = objectMapperConfiguration.extensionPluginConfigObjectMapper();
+        final TestType duration = objectMapper.convertValue(testString, TestType.class);
+        assertThat(duration, equalTo(TestType.fromOptionValue(testString)));
+    }
+
+    private enum TestType {
+
+        TEST("test");
+
+        private static final Map<String, TestType> NAMES_MAP = Arrays.stream(TestType.values())
+                .collect(Collectors.toMap(TestType::toString, Function.identity()));
+
+        private final String name;
+
+        TestType(final String name) {
+            this.name = name;
+        }
+
+        public String toString() {
+            return this.name;
+        }
+
+        @JsonCreator
+        static TestType fromOptionValue(final String option) {
+            return NAMES_MAP.get(option);
+        }
+    }
+
+}

